src/main/kotlin/app/ui/profilePreview/ViewModel/ProfilePreviewViewModel.kt

```diff
@@ -0,0 +1,36 @@
+package app.ui.profilePreview.ViewModel
+
+import app.ui.profilePreview.Model.ProfilePreviewModel
+import io.reactivex.subjects.PublishSubject
+
+class ProfilePreviewViewModel(
+        private val userIconHash: PublishSubject<String>,
+        private val clickNext: PublishSubject<Boolean>,
+        private val clickRedo: PublishSubject<Boolean>,
+        private val userListenedAudio: PublishSubject<Boolean>
+
+) {
+
+    private val model = ProfilePreviewModel()
+
+    fun newIconHash(iconHash: String) {
+        userIconHash.onNext(iconHash)
+        model.userIconHash = iconHash
+    }
+
+    fun clickRedo(clicked: Boolean) {
+        clickRedo.onNext(clicked)
+        model.clickRedo = clicked
+    }
+
+    fun clickNext(clicked: Boolean) {
+        clickNext.onNext(clicked)
+        model.clickNext = clicked
+    }
+
+    fun listenedAudio(listened: Boolean) {
```

Comment:
this is a really awkward naming convention. I know what you're trying to communicate but it's not obvious at first. Not sure what yet, but let's try to think of a different way of phrasing this before committing it.